src/network/networkd-manager.c

```diff
@@ -558,34 +558,33 @@ int manager_rtnl_process_address(sd_netlink *rtnl, sd_netlink_message *message,
         if (r < 0 && r != -ENODATA) {
                 log_link_warning_errno(link, r, "rtnl: cannot get IFA_CACHEINFO attribute, ignoring: %m");
                 return 0;
-        } else if (r >= 0) {
+        } else if (r >= 0)
                 if (cinfo.ifa_valid != CACHE_INFO_INFINITY_LIFE_TIME)
                         valid_str = format_timespan(valid_buf, FORMAT_TIMESPAN_MAX,
                                                     cinfo.ifa_valid * USEC_PER_SEC,
                                                     USEC_PER_SEC);
-        }
 
         (void) address_get(link, family, &in_addr, prefixlen, &address);
 
         switch (type) {
         case RTM_NEWADDR:
                 if (address)
-                        log_link_debug(link, "Updating address: %s/%u (valid %s%s)", buf, prefixlen,
+                        log_link_debug(link, "Updating address: %s/%hhu (valid %s%s)", buf, prefixlen,
```

Comment:
hmm, interesting. I wasn't aware of "hh" I must admit. And I am puzzled by it. I was sure there was the rule in C that integers shorter than "int"/"unsigned int" get expanded to "int"/"unsigned int" anyway when passed via varargs. So what does "hh" actually provide us with?

Reply:
Hmm, there's a discussion here:

https://stackoverflow.com/questions/4586962/what-is-the-purpose-of-the-h-and-hh-modifiers-for-printf

It's inconclusive, the way I read it?

i am tempted to say, adding "hh" everywhere is probably not worth the trouble? Or am I missingsomething?

Reply:
OK. The changes are dropped at least from this PR.